### ISSUE TITLE
Same output from sigmoid function at merge layer in Siamese network

### ISSUE THREAD

**Issue by NidhiRustagi** [NidhiRustagi/LSTM-networks-for-evaluating-descriptive-answers#1]
My project aims to automatically evaluate short descriptive answers i.e. to assign numerical score between 0 and 1 to a given student answer. The basic logic is compare student given answer with faculty given modal answer i.e. finding document similarity where document length as of now is restricted to 2-3 sentences. 

1) convert student answer and faculty answer using word2vec embeddings.
2) this embedding is used as left and right input to Siamese architecture
3) apply lstm on this embedding layer
3) merge the two networks using dot merge function
4) Apply sigmoid activation function to get numerical score at the end

Here are some of my doubts
a) For training the word2vec model, an input file with 70 lines and 1,300 words. Is this sufficient to train a word2vec model?
b) Training data consists of 25 sentences. is this too low  value for training the weights of the neural network?
c) Why does my cosine similarity function return high values even for highly unlikely pair of sentences.

```
modal_ans = "An operating system acts as an interface between the user and computer hardware. Operating system controls and coordinates the use of computing resources among the various application programs. Some examples of operating systems are windows, linux, unix, android and ios."

ansvector = getWord2Vec(w2v_model,modal_ans)

st_ans = "hardware acts as an interface between operating system and software"
stvector = getWord2Vec(w2v_model,st_ans)
np.array_equal(ansvector, stvector)
similarity(ansvector,stvector)
```
Output : 0.930095 which is a very high cosine similarity score.

This is the first neural network project using Keras. I did try to google the error also but have not been very successful. Any help will be greatly appreciated.
